tests/rptest/tests/fetch_fairness_test.py

```diff
@@ -89,15 +90,26 @@ def multiple_partitions(count):
         # configure kcl to fetch at most 1 byte in single fetch request,
         # this way we should receive exactly one record per each partition
 
+        consumed_frequency = defaultdict(lambda: 0)
         kcl = KCL(self.redpanda)
+
+        expected_frequency = 5
+        # issue single kcl command that will generate multiple fetch requests
+        # in single fetch session, it should include single partition in each
+        # fetch response.
+
         consumed = kcl.consume(topic="topic-.*",
-                               n=number_of_partitions,
+                               n=number_of_partitions * expected_frequency,
                                regex=True,
                                fetch_max_bytes=1,
                                group="test-gr-1")
-        consumed_partitions = set()
+
         for c in consumed.split():
             parts = c.split('.')
-            consumed_partitions.add((parts[1], parts[2]))
+            consumed_frequency[(parts[1], parts[2])] += 1
 
-        assert len(consumed_partitions) == number_of_partitions
+        for p, freq in consumed_frequency.items():
+            self.redpanda.logger.info(f"consumed {freq} messages from: {p}")
+            # assert that frequency is in expected range
+            assert freq <= expected_frequency + 1
```

Comment:
Maybe instead of `1` use expected_frequency*0.1, so make it more adaptable to parameter changes.

I haven't actually looked at how fairness works in clients so I'm assuming it's somewhat sloppy